_pytest/skipping.py

```diff
@@ -147,10 +148,17 @@ def getexplanation(self):
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_runtest_setup(item):
-    evalskip = MarkEvaluator(item, 'skipif')
-    if evalskip.istrue():
-        item._evalskip = evalskip
-        pytest.skip(evalskip.getexplanation())
+    eval_skipif = MarkEvaluator(item, 'skipif')
+
+    if eval_skipif.istrue():
+        item._evalskip = eval_skipif
+        pytest.skip(eval_skipif.getexplanation())
+    elif isinstance(item.keywords.get('skip'), MarkInfo):
```

Comment:
Hmm I noticed `reason` is not handled here, and the user could still use `pytest.mark.skip('sys.version_info[0] == 2')` and the test would not be skipped on python 2. This fixes it, but it is not pretty:

```python
@pytest.hookimpl(tryfirst=True)
def pytest_runtest_setup(item):
    evalskip = MarkEvaluator(item, 'skip')
    if evalskip.holder is not None:
        evalskip.reason = evalskip.holder.kwargs.get('reason', 'explicit skip')
        evalskip.result = True
        item._evalskip = evalskip
        pytest.skip(evalskip.getexplanation())

    evalskip = MarkEvaluator(item, 'skipif')
    if evalskip.istrue():
        item._evalskip = evalskip
        pytest.skip(evalskip.getexplanation())
    ...
```